src/DepositManager.mo

```diff
@@ -198,33 +167,37 @@ module {
         });
         case (#err err) #consolidationError(err);
       };
-      log(p, event);
+      log(entry.key(), event);
 
       // process result
       switch (res) {
         case (#ok _) {
           totalConsolidated += consolidated;
-          credit.changePool(consolidated - credited);
-          ignore release(null);
+          data.pool += surcharge;
         };
         case (#err _) {
-          credit.changeUser(p, -credited);
-          // the fees may have changed in the meantime, hence we process the exising deposit as a new one
-          ignore process_deposit(p, deposit, release);
+          // also adds to depositsTree
+          entry.setDeposit(deposit);
         };
       };
 
+      underwayFunds -= deposit;
+
+      assert entry.unlock();
```

Comment:
We are double-unlocking here. We do it here inside `consolidate()`. But the lock was obtained outside of `consolidate()` and is released outside as well. For example in trigger() the code is:
```
        assert entry.lock();
        let result = await* consolidate(entry);
        assert entry.unlock();
   ```